compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -912,11 +912,44 @@ ofInterestRule opts = do
             garbageCollect (`HashSet.member` reachableFiles)
           DamlEnv{..} <- getDamlServiceEnv
           liftIO $ whenJust envScenarioService $ \scenarioService -> do
-              ctxRoots <- modifyVar envScenarioContexts $ \ctxs -> do
-                  let gcdCtxs = HashMap.filterWithKey (\k _ -> k `HashSet.member` roots) ctxs
-                  pure (gcdCtxs, HashMap.elems gcdCtxs)
-              prevCtxRoots <- modifyVar envPreviousScenarioContexts $ \prevCtxs -> pure (ctxRoots, prevCtxs)
-              when (prevCtxRoots /= ctxRoots) $ void $ SS.gcCtxs scenarioService ctxRoots
+              mask $ \restore -> do
```

Comment:
IIRC there were some issues around `mask` vs. `uninterruptibleMask` with GRPC. What's the reasoning for `mask` instead of `uninterruptibleMask` here?

Reply:
I somewhat wondered about this as well. For the other grpc issue, we ended up refactoring it to be save even if `mask` is interrupted instead of using `uninterruptibleMask` (which seems a bit fishy for a network request). I believe that is the case here as well. If we don’t end up GCing we will restore the set of context roots to the old roots so at worst we will GC again.

Reply:
That seems reasonable. Thanks for explaining.